homeassistant/components/tuya/entity.py

```diff
@@ -256,10 +256,40 @@ def get_dptype(
             order = ["function", "status_range"]
         for key in order:
             if dpcode in getattr(self.device, key):
-                return DPType(getattr(self.device, key)[dpcode].type)
+                try:
+                    return DPType(getattr(self.device, key)[dpcode].type)
+                except ValueError:
+                    fixed_type = TuyaEntity.determine_dptype(
+                        getattr(self.device, key)[dpcode].type
+                    )
+                    LOGGER.warning(
+                        f'Device {self.device.name} (id: {self.device.id}) has returned a bad model dpId type for code {dpcode} : "{getattr(self.device, key)[dpcode].type}", it should have been "{fixed_type}". Please contact the Tuya Support about this'
+                    )
+                    return fixed_type
 
         return None
 
+    @staticmethod
+    def determine_dptype(type) -> DPType | None:
+        """Determine the DPType.
+
+        Sometimes, we get ill formed DPTypes from the cloud,
+        this fixes them and maps them to the correct DPType
+        """
+        if type == "value":
+            return DPType(DPType.INTEGER)
```

Comment:
Why create another instance?

```suggestion
            return return DPType.INTEGER
```